app/models/task.py

```diff
@@ -1,5 +1,37 @@
 from app import db
+import datetime
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    is_complete = db.Column(db.Boolean, default=False)
+    goal_id = db.Column(db.Integer, db.ForeignKey("goal.goal_id"), nullable=True)
+    goal = db.relationship("Goal", back_populates="tasks", lazy=True)
+
+    def to_dict(self):
+        task_dict = {}
+        if self.goal_id is not None:
+            task_dict = {
+                "id": self.task_id,
+                "goal_id": self.goal_id,
+                "title": self.title,
+                "description": self.description,
+                "is_complete": self.is_complete
+            } 
+        else:
+            task_dict = {
+                "id": self.task_id,
+                "title": self.title,
+                "description": self.description,
+                "is_complete": self.is_complete
+            }
+        return task_dict
```

Comment:
This seems a little bigger than it needs to be.  Could you make the basic dictionary then add key-value pairs like goal_id as needed?